BitList.cs

```diff
@@ -0,0 +1,1553 @@
+    {
+        left.Or(right);
+        return left;
+    }
+
+    public static BitList operator ^(BitList left, BitList right)
+    {
+        left.Xor(right);
+        return left;
+    }
+
+    public static BitList operator +(BitList left, BitList right)
+    {
+        var leftBitList = left.ToBooleanArray();
+        var rightBitList = right.ToBooleanArray();
+        var leftList = leftBitList.ToList();
+        foreach (var bit in rightBitList)
+            leftList.Add(bit);
+        return new BitList(leftList.ToArray());
+    }
+
+    public static BitList operator +(BitList left, bool right)
+    {
+        return left + new BitList(right);
+    }
+
+    public static BitList operator +(bool left, BitList right)
+    {
+        return right + left;
+    }
+
+    public static bool operator ==(BitList left, BitList right)
+    {
+        return left.Equals(right);
+    }
+
+    public static bool operator !=(BitList left, BitList right)
+    {
+        return !left.Equals(right);
+    }
+
+    public static BitList operator <<(BitList left, int right)
+    {
+        left.LeftShift(right);
+        return left;
+    }
+
+    public static BitList operator >>(BitList left, int right)
+    {
+        left.RightShift(right);
+        return left;
+    }
+
+    public static explicit operator bool[](BitList value)
+    {
+        return value.ToBooleanArray();
+    }
+
+    public static explicit operator byte[](BitList value)
+    {
+        return value.ToByteArray();
+    }
+
+    public static explicit operator short[](BitList value)
+    {
+        return value.ToInt16Array();
+    }
+
+    public static explicit operator int[](BitList value)
+    {
+        return value.ToInt32Array();
+    }
+
+    public static explicit operator long[](BitList value)
+    {
+        return value.ToInt64Array();
+    }
+
+    public static explicit operator sbyte[](BitList value)
+    {
+        return value.ToSByteArray();
+    }
+
+    public static explicit operator ushort[](BitList value)
+    {
+        return value.ToUInt16Array();
+    }
+
+    public static explicit operator uint[](BitList value)
+    {
+        return value.ToUInt32Array();
+    }
+
+    public static explicit operator ulong[](BitList value)
+    {
+        return value.ToUInt64Array();
+    }
+
+    public static explicit operator float[](BitList value)
+    {
+        return value.ToSingleArray();
+    }
+
+    public static explicit operator double[](BitList value)
+    {
+        return value.ToDoubleArray();
+    }
+
+    public static explicit operator decimal[](BitList value)
+    {
+        return value.ToDecimalArray();
+    }
+
+    public static explicit operator char[](BitList value)
+    {
+        return value.ToCharArray();
+    }
+
+    public static explicit operator string(BitList value)
+    {
+        return value.ToString();
+    }
+
+    private class BitListEnumerator : IEnumerator<bool>, IDisposable
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fix this implementation of 'IDisposable' to conform to the dispose pattern.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5545753)